e2e/utils/process-utils.ts

```diff
@@ -39,3 +39,19 @@ export async function killPorts(port?: number): Promise<boolean> {
     ? await killPort(port)
     : (await killPort(3333)) && (await killPort(4200));
 }
+
+export async function killProcessAndPorts(
+  pid: number | undefined,
+  ...ports: number[]
+): Promise<void> {
+  try {
+    if (pid) {
+      await promisifiedTreeKill(pid, 'SIGKILL');
+    }
+    for (const port of ports) {
+      await killPort(port);
+    }
+  } catch (err) {
+    expect(err).toBeFalsy();
```

Comment:
We want to catch an error, print it out and fail test rather than have exception blow up, since it's better for the readability and subsequent debugging.

This way failed tests have the same output format as those that had an exception during the run.

Apart from that, there is no other reason. That assertion could be anything as long as it reports a mismatch and shows an entire error with the stack trace.

Reply:
If you find a better way to do this, feel free to improve it everywhere

Reply:
Sounds good, I'll check whether it makes sense to do something else, but it's not a priority anyways. This is just a cleanup, not an actual assertion or the focus of the tests.

Reply:
cool, makes sense just wanted to make sure it wasn't supposed to be changed/updated for any reason.